app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb

```diff
@@ -55,18 +55,27 @@ def refresh_ems
   end
 
   def update_with_provider_object(raw_job)
-    self.ems_ref     ||= raw_job.id
-    self.status        = raw_job.status
-    self.start_time  ||= raw_job.started
-    self.finish_time ||= raw_job.finished
-    self.verbosity   ||= raw_job.verbosity
+    update_attributes(
+      :ems_ref     => raw_job.id,
+      :status      => raw_job.status,
+      :start_time  => raw_job.started,
+      :finish_time => raw_job.finished,
+      :verbosity   => raw_job.verbosity
+    )
 
     if parameters.empty?
       self.parameters = raw_job.extra_vars_hash.collect do |para_key, para_val|
         OrchestrationStackParameter.new(:name => para_key, :value => para_val, :ems_ref => "#{raw_job.id}_#{para_key}")
       end
     end
-    save!
+
+    if authentications.empty?
+      self.authentications = %w(credential_id cloud_credential_id network_credential_id).collect do |credential_attr|
+        credential_ref = raw_job.try(credential_attr)
+        next if credential_ref.blank?
+        ext_management_system.credentials.find_by(:manager_ref => credential_ref)
```

Comment:
Yeah, I understand that...still not sure how that applies to my comment...

find_by in a loop is an N+1 and can be easily fixed with collecting the query criteria up front.